### ISSUE TITLE
Event triggered by [harm_unit]

### ISSUE THREAD

**Issue by rodolphecombe** [wesnoth/wesnoth#8974]
### Describe the desired feature

There is currently no way to trigger an event with a [harm_unit] if the impacted unit is not killed. This is the only "default" way to inflict damage to a unit (the other two being attacker hits and defender hits) that is not able to trigger an event (the other two have their own). 

The FR would be to add a unit_harmed,  or whatever the name, event that would trigger either only on harm _unit, or on all three cases. One could even imagine to add both. 

Use case is for example events that need to trigger when a unit is losing HP, or that are tracking units HP. 

There are of course other ways to change unit HP such as [modify_unit] but it might be too complex to also trigger on such occasions.

**Comment by ProditorMagnus** [comment]
Are you asking something you cant already do in your addon by overriding wesnoth.wml_actions.harm_unit?

**Comment by rodolphecombe** [comment]
> Are you asking something you cant already do in your addon by overriding wesnoth.wml_actions.harm_unit?

The add-on author might not be the one having written the [harm _unit]. It could be one from mainline interacting with the add-on.

**Comment by ProditorMagnus** [comment]
If you override wml tag then everyone who calls it, including mainline, uses your version of tag.

**Comment by rodolphecombe** [comment]
> If you override wml tag then everyone who calls it, including mainline, uses your version of tag. 

That is actually quite interesting. I will try to make a [harm _unit] that triggers a on_harm event and will close the FR by posting it if I am successful.

**Comment by CelticMinstrel** [comment]
I don't think it makes sense to have an event unique to `[harm_unit]`, though if you need one in your add-on, then of course that's fine to implement it in your add-on.

**Comment by rodolphecombe** [comment]
I made a first PR for this issue (https://github.com/wesnoth/wesnoth/pull/8995), adding a pre harm and a post harm event. 
After discussing with Ravana, I think the two custom events could be replacedby pre attack and attacker/defender/unit hits events.
Do you have any opinion on that?

**Comment by Pentarctagon** [comment]
What would a pre-harm event be? pre attack would be before you know if the unit will be harmed, wouldn't it?

**Comment by rodolphecombe** [comment]
ATM pre harm tells that the unit is the target of a [harm_unit].
It indeed seems that attack event would be more suitable than pre attack then. 
What do you think?

**Comment by ProditorMagnus** [comment]
I think you should name the event how you need for your addon and not try to get it to mainline.
